### ISSUE TITLE
need some help

### ISSUE THREAD

**Issue by xialisuper** [GIfatahTH/states_rebuilder#36]
hi, love the package.
im new to flutter and trying to manage the statement . I have tried bloc ,provider,  and states rebuilder. Can you help me to use this package in a larger project? I don't real know how to keep the code 'clean' . eg :  I get a ReactModel to control a page, so there are methods to do the business logic ,get and set data , and so on. finally I made the ReactModel such a mess. Can you just give me some ideas to solve this problem?

**Comment by GIfatahTH** [comment]
What you need to make your code clean is  The Clean Architecture.
There is a good playlist on youtube by ResoCoder about using the Clean architecture with Test Driven Design. Please refer to it and learn the best practices:
https://www.youtube.com/watch?v=dc3B_mMrZ-Q&t=6977s

Here is my implementation of clean architecture using states_rebuilder.
https://github.com/GIfatahTH/states-rebuilder-examples/tree/master/007-clean_architecture_dane_mackier_app

Please report an issue there, if you do not understand something.

**Comment by xialisuper** [comment]
Thanks for the examples. They are really helpful. 
And I got another problem here. 
My app is like this. 
Material App :
- BottomTabBar
-- Page1_1 -> Page1_2
-- Page2_1 -> Page2_2 -> Page2_3 -> ... 
-- Page3_1 -> Page3_2

I need a Service to be reachable through out "-- Page2_1 -> Page2_2 -> Page2_3 -> ... "
I read  about the "reInject" , so I did just like that when I push to next page and it works . But when the pages become more and more . I realize I could be doing it  the wrong way.
I know that I can inject the Service to MaterialApp and it will keep alive when App is active.
But in this case , I don't need the Service to be reached by Page1 or Page3 . What can I do to solve this problem.
